lib/block_view.go

```diff
@@ -4726,135 +5590,780 @@ func (bav *UtxoView) _connectUpdateProfile(
+	bav._setProfileEntryMappings(&newProfileEntry)
+
+	// Add an operation to the list at the end indicating we've updated a profile.
+	utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+		Type:             OperationTypeUpdateProfile,
+		PrevProfileEntry: prevProfileEntry,
+	})
+
+	return totalInput, totalOutput, utxoOpsForTxn, nil
+}
+
+func (bav *UtxoView) _connectCreateNFT(
+	txn *MsgBitCloutTxn, txHash *BlockHash, blockHeight uint32, verifySignatures bool) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+
+	// Check that the transaction has the right TxnType.
+	if txn.TxnMeta.GetTxnType() != TxnTypeCreateNFT {
+		return 0, 0, nil, fmt.Errorf("_connectCreateNFT: called with bad TxnType %s",
+			txn.TxnMeta.GetTxnType().String())
+	}
+	txMeta := txn.TxnMeta.(*CreateNFTMetadata)
+
+	// Validate the txMeta.
+	if txMeta.NumCopies > bav.GlobalParamsEntry.MaxCopiesPerNFT {
+		return 0, 0, nil, RuleErrorTooManyNFTCopies
+	}
+	if txMeta.NumCopies == 0 {
+		return 0, 0, nil, RuleErrorNFTMustHaveNonZeroCopies
+	}
+	royaltyBasisPoints := txMeta.NFTRoyaltyToCreatorBasisPoints + txMeta.NFTRoyaltyToCoinBasisPoints
+	if royaltyBasisPoints > bav.Params.MaxNFTRoyaltyBasisPoints {
+		return 0, 0, nil, RuleErrorNFTRoyaltyHasTooManyBasisPoints
+	}
+	postEntry := bav.GetPostEntryForPostHash(txMeta.NFTPostHash)
+	if postEntry == nil || postEntry.isDeleted {
+		return 0, 0, nil, RuleErrorCreateNFTOnNonexistentPost
+	}
+	if IsVanillaReclout(postEntry) {
+		return 0, 0, nil, RuleErrorCreateNFTOnVanillaReclout
+	}
+	if !reflect.DeepEqual(postEntry.PosterPublicKey, txn.PublicKey) {
+		return 0, 0, nil, RuleErrorCreateNFTMustBeCalledByPoster
+	}
+	if postEntry.IsNFT {
+		return 0, 0, nil, RuleErrorCreateNFTOnPostThatAlreadyIsNFT
+	}
+	profileEntry := bav.GetProfileEntryForPublicKey(postEntry.PosterPublicKey)
+	if profileEntry == nil || profileEntry.isDeleted {
+		return 0, 0, nil, RuleErrorCantCreateNFTWithoutProfileEntry
+	}
+
+	// Connect basic txn to get the total input and the total output without
+	// considering the transaction metadata.
+	totalInput, totalOutput, utxoOpsForTxn, err := bav._connectBasicTransfer(
+		txn, txHash, blockHeight, verifySignatures)
+	if err != nil {
+		return 0, 0, nil, errors.Wrapf(err, "_connectCreateNFT: ")
+	}
+
+	// Force the input to be non-zero so that we can prevent replay attacks.
+	if totalInput == 0 {
+		return 0, 0, nil, RuleErrorCreateNFTRequiresNonZeroInput
+	}
+
+	if verifySignatures {
+		// _connectBasicTransfer has already checked that the transaction is
+		// signed by the top-level public key, which we take to be the poster's
+		// public key.
+	}
+
+	// Since issuing N copies of an NFT multiplies the downstream processing overhead by N,
+	// we charge a fee for each additional copy minted.
+	nftFee := txMeta.NumCopies * bav.GlobalParamsEntry.CreateNFTFeeNanos
+
+	// Ensure that the transaction covers the NFT fee.
+	totalOutput += nftFee
```

Comment:
Out of pure paranoia, I would add an overflow check here. I like to do it after every + operation.